src/main/java/bio/terra/service/filedata/DrsService.java

```diff
@@ -378,11 +378,17 @@ private DRSAccessURL signAzureUrl(
 
   private DRSAccessURL signGoogleUrl(
       SnapshotCacheResult cachedSnapshot, String gsPath, AuthenticatedUserRequest authUser) {
-    Storage storage =
-        StorageOptions.newBuilder()
-            .setProjectId(cachedSnapshot.googleProjectId)
-            .build()
-            .getService();
+    Storage storage;
+    if (cachedSnapshot.isSelfHosted) {
```

Comment:
Would it be worth adding a unit test for this case? It looks like it could be tested by adding another test case for `lookupSnapshotForDRSObject()`.

Reply:
I was debating.  Mocking out credentials gets weird.  And the true test is whether you can access the file...it's tough because the failure was ultimately outside of our boundary of control